components/renku_data_services/secrets/db.py

```diff
@@ -53,16 +53,35 @@ async def get_secrets_by_ids(self, requested_by: APIUser, secret_ids: list[str])
             orms = res.scalars()
             return [orm.dump() for orm in orms]
 
+    @staticmethod
+    def create_secret_orm(
+        name: str, user_id: str, encrypted_value: bytes, encrypted_key: bytes, kind: SecretKind
+    ) -> SecretORM:
+        """Create a new secret."""
+        return SecretORM(
+            name=name,
+            modification_date=datetime.now(UTC).replace(microsecond=0),
+            user_id=user_id,
+            encrypted_value=encrypted_value,
+            encrypted_key=encrypted_key,
+            kind=kind,
+        )
+
+    @staticmethod
+    def update_secret_orm(secret: SecretORM, encrypted_value: bytes, encrypted_key: bytes) -> None:
+        """Update an existing secret."""
+        secret.encrypted_value = encrypted_value
+        secret.encrypted_key = encrypted_key
+        secret.modification_date = datetime.now(UTC).replace(microsecond=0)
+
```

Comment:
Suggestion: I would put these on SecretORM directly, the first as a class method, the latter just regular method.